### DISCUSSION TITLE
Annotate by match order

### DISCUSSION THREAD

**Discussion by mdorkhah** [GateNLP/python-gatenlp#95] [Help Q&A]
How can I separately annotate each result?
For example annotating the first match to type1, second match to type2, third match to type3 and so on...

**Comment by johann-petrak** [comment]
Can you give a more concrete example?

If you have a fixed number of matches, then you can simply give each match a different name and use that name in different AddAnn actions. 

If the number of matches is unknown in advance, you would need to implement your own action which goes through the matching data. 
Please note that some names of the matching data API will change in the next release. 

Hopefully there will be more detailed documentation about this eventually, but in general, the planned nomenclature for these more complex situations will be:
* a `match` is where a named pattern fits your document, a successful result can have many such matches. If a pattern fits your document, no match information will be created for performance reasons
* a `result` is a successful way to match patterns against your document. With some patterns and documents, it is possible to have several results because annotations can arbitrarily overlap, so e.g. the same range in a document could match a sequence of typeA typeB type A, or else a sequnce of typeC typeD which may be Or alternatives in the pattern.

**Reply by mdorkhah** [reply]
> Can you give a more concrete example?
> 
> If you have a fixed number of matches, then you can simply give each match a different name and use that name in different AddAnn actions.
> 
> If the number of matches is unknown in advance, you would need to implement your own action which goes through the matching data.
> Please note that some names of the matching data API will change in the next release.
> 
> Hopefully there will be more detailed documentation about this eventually, but in general, the planned nomenclature for these more complex situations will be:
> 
> * a `match` is where a named pattern fits your document, a successful result can have many such matches. If a pattern fits your document, no match information will be created for performance reasons
> * a `result` is a successful way to match patterns against your document. With some patterns and documents, it is possible to have several results because annotations can arbitrarily overlap, so e.g. the same range in a document could match a sequence of typeA typeB type A, or else a sequnce of typeC typeD which may be Or alternatives in the pattern.

Consider this text "..., Sport: Soccer,..., Sport: Swimming,..., Sport: Wrestling, ..."
If I want to match the sport (from list) which comes after term "Sports:" and annotate first result, which is Soccer to type1, second result Swimming to type2, third result Wrestling to type3 and so on. The number of results is unknown.
